molecule/__init__.py

```diff
@@ -18,8 +18,20 @@
 #  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 #  THE SOFTWARE.
 
+import signal
+import sys
+
 from pbr import version
 
+
+def signal_handler(signal, frame):
+    msg = '{red}Exiting molecule on Ctrl+c.{reset}'
+    print(msg.format(red='\033[31m', reset='\033[0m'))
+    sys.exit()
```

Comment:
As far as I can tell, once the handler is registered, it will be caught even if we're loading other modules after it. It seems to benefit from being as early in the code path as possible, which is why I was trying not to bury it under a bunch of includes.